File: src/main/java/com/gimranov/zandy/app/model/ApiResponse.java

package com.gimranov.zandy.app.model;

import java.io.Serializable;
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import lombok.Data;

@Data
public class ApiResponse<T> implements Serializable
{

    @SerializedName("key")
    @Expose
    public String key;
    @SerializedName("version")
    @Expose
    public int version;
    @SerializedName("library")
    @Expose
    public Library library;
    @SerializedName("links")
    @Expose
    public Links links;
    @SerializedName("meta")
    @Expose
    public Meta meta;
    @SerializedName("data")
    @Expose
    public T data;
    private final static long serialVersionUID = -8693618652266258725L;

}

File: src/main/java/com/gimranov/zandy/app/model/ApiResponse.kt
package com.gimranov.zandy.app.model

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

data class ApiResponse<T>(

        @SerializedName("key")
        @Expose
        var key: String? = null,
        @SerializedName("version")
        @Expose
        var version: Int = 0,
        @SerializedName("library")
        @Expose
        var library: Library? = null,
        @SerializedName("links")
        @Expose
        var links: Links? = null,
        @SerializedName("meta")
        @Expose
        var meta: Meta? = null,
        @SerializedName("data")
        @Expose
        var data: T? = null
)
